isis/tests/FunctionalTestsCubeatt.cpp

```diff
@@ -0,0 +1,71 @@
+#include <iostream>
+
+#include "Cube.h"
+#include "Fixtures.h"
+
+#include "cubeatt.h"
+#include "gmock/gmock.h"
+
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/cubeatt.xml").expanded();
+
+// Tests setting output attributes: bit type and range
+TEST_F(SmallCube, FunctionalTestCubeattBitttypeAndRange) {
+  QString cubePath = tempDir.path() + "/bitTypeCubeatt.cub+8bit+0.0:1.0";
+
+  QVector<QString> args = {"from=" + testCube->fileName(), "to=" + cubePath};
+  UserInterface options(APP_XML, args);
+  cubeatt(options);
+
+  // Test the label
+  Cube outputCube(cubePath);
+  Pvl *label = outputCube.label();
+  PvlGroup &pixelGroup = label->findObject("IsisCube").findObject("Core").findGroup("Pixels");
+  EXPECT_EQ(QString(pixelGroup["Type"]).toStdString(), "UnsignedByte");
+  EXPECT_NEAR(pixelGroup["Base"][0].toDouble(), -0.003952569, 0.00001);
+  EXPECT_NEAR(pixelGroup["Multiplier"][0].toDouble(), 0.003952569, 0.00001);
+
+  // Test the DNs
+}
+
+TEST_F(SmallCube, FunctionalTestCubeattNoChange) {
+  QString cubePath = tempDir.path() + "/NoChangeCubeatt.cub";
+  QVector<QString> args = {"from=" + testCube->fileName() + "+1", "to=" + cubePath};
+  UserInterface options(APP_XML, args);
+  cubeatt(options);
+
+  // Test the label
+  Cube outputCube(cubePath);
+  Pvl *label = outputCube.label();
+  PvlGroup &pixelGroup = label->findObject("IsisCube").findObject("Core").findGroup("Pixels");
+  EXPECT_EQ(QString(pixelGroup["Type"]).toStdString(), "Real");
+  EXPECT_EQ(pixelGroup["Base"][0].toDouble(), 0);
+  EXPECT_EQ(pixelGroup["Multiplier"][0].toDouble(), 1);
```

Comment:
Again use the accessors, don't parse the label text. You should check all of the attributes of the cubes:

* pixel type
* storage format
* label format
* storage order
* pixel range
* bands

Reply:
I always forget I can just do this. Thanks, @jessemapel

Reply:
When set, the pixel range is just used by the cube class to set the base and multiplier, so checking those is a way to check that the min/max was set appropriately.